### ISSUE TITLE
4.x: Make sure all threads are named

### ISSUE THREAD

**Issue by barchetta** [helidon-io/helidon#5359]
In 2.x we made an effort to make sure all threads are named (https://github.com/helidon-io/helidon/issues/4159,  https://github.com/helidon-io/helidon/issues/3784). Looking at the 4.x code it doesn't look like we are naming VirtualThreads:

```
./webclient/webclient/src/main/java/io/helidon/nima/webclient/LoomClient.java:        return Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
./webserver/webserver/src/main/java/io/helidon/nima/webserver/ServerListener.java:        this.readerExecutor = Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
./webserver/webserver/src/main/java/io/helidon/nima/webserver/ServerListener.java:        this.sharedExecutor = Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
./webserver/webserver/src/main/java/io/helidon/nima/webserver/ConnectionHandler.java:            //                                                         Thread.ofVirtual().factory(), (scope, future) -> {})) {
./webserver/webserver/src/main/java/io/helidon/nima/webserver/LoomServer.java:        this.executorService = Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
./fault-tolerance/src/main/java/io/helidon/nima/faulttolerance/AsyncImpl.java:        this.executor = LazyValue.create(Executors.newVirtualThreadPerTaskExecutor());
./fault-tolerance/src/main/java/io/helidon/nima/faulttolerance/AsyncImpl.java:                : LazyValue.create(Executors.newVirtualThreadPerTaskExecutor());
```

We should ensure all threads, virtual or not, are named with something useful. 

We should also keep an eye on the JDK efforts for organizing and displaying thread dumps for virtual threads to see if there is anything else we should do in this area.

**Comment by tomas-langer** [comment]
1. Some threads are named from within the threads (as we may use the socket id).
2. Structured concurrency is on the radar - there were too big changes in the API when Nima was created, let's wait for that feature to settle

**Comment by tomas-langer** [comment]
Resolved by #5397
